ExternalLookup/src/main/java/cheetah/example/externallookup/function/ExternalLookupMapper.java

```diff
@@ -51,6 +56,15 @@ public void asyncInvoke(InputEvent externalLookupInputEvent, ResultFuture<Output
     public void open(Configuration parameters) {
         client = HttpClient.newHttpClient();
 
+        tokenProvider = KeyedTokenProvider.getInstance();
+        tokenProvider.registerTokenProviderIfAbsent(TOKEN_ID,
+                () -> new CachedTokenProvider(
+                        new OAuthTokenProvider(
+                                System.getenv("SERVICE_TOKEN_ENDPOINT"),
+                                System.getenv("SERVICE_CLIENT_ID"),
+                                System.getenv("SERVICE_CLIENT_SECRET"),
+                                System.getenv("SERVICE_SCOPE"))));
+
         ParameterTool parameterTool = (ParameterTool)
                 getRuntimeContext().getExecutionConfig().getGlobalJobParameters();
         idServiceHost = parameterTool.get("id-service-url");
```

Comment:
i would create a field with this and set it in the constructor so the main job class is responsible for getting program arguments and the ExternalLookupMapper becomes more modular